### PR TITLE
Task detail view

### PR DESCRIPTION
**Pull request by alex6296**
whould like some reivew already now this thing is fat

### CODE REVIEW

File: src/api/messages.js [hunk 1]
```diff
@@ -16,8 +16,8 @@ export default {
    * @param {Number} id
    * @return {Promise}
    */
-  async get(id) {
-    const res = await request.get(id ? `messages/${id}` : 'messages')
+  async get(taskId) {
+    const res = await request.get(`messages/${taskId}`)
```
Comment (kvartborg) [inline src/api/messages.js]: This change should not be there

File: src/api/offer.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import request from './request'
+import { struct } from 'superstruct'
+
+/**
+ * Defines input-types required for validation
+ */
```
Comment (kvartborg) [inline src/api/offer.js]: Delete this file

File: src/views/Task/View.js [hunk 3]
```diff
@@ -1,64 +1,159 @@
...
+   * adds changes to this.state
+   * @param  {Object} event
+   */
+  changeHandler = event => {
+      const tempOffer = JSON.parse(JSON.stringify(this.state.offer))
+      tempOffer[event.target.name] = event.target.value
+      this.setState({
+         offer:tempOffer
+      })
+    }
```
Comment (kvartborg) [inline src/views/Task/View.js]: Indentation

File: src/views/Task/View.js [hunk 4]
```diff
@@ -1,64 +1,159 @@
...
+        currency:'',
+        complexity:'',
+      },
+      error:''
+    }
+  }
 
-    this.loadTasks(this.props.match.params.id);
-  };
+  componentDidMount(){
```
Comment (grammeaway) [inline src/views/Task/View.js]: Needs docblock

File: src/views/Task/View.js [hunk 5]
```diff
@@ -1,64 +1,162 @@
...
+   * @param {Object} event
+   */
+  submitHandler= async event =>{
+    event.preventDefault();
+    try {
+      const res = await Task.createEstimate( this.state.task.id,{
+      price: Number(this.state.offer.price),
+      totalHours:Number(this.state.offer.totalHours),
+      currency: String(await currencies.get()),
+      complexity:Number(this.state.offer.complexity),
```
Comment (kvartborg) [inline src/views/Task/View.js]: Indent

File: src/views/Task/View.js [hunk 6]
```diff
@@ -1,64 +1,162 @@
...
+  }
+
+  /**
+   * event listener for input
+   * adds changes to this.state
+   * @param  {Object} event
+   */
+  changeHandler = event => {
+    const tempOffer = JSON.parse(JSON.stringify(this.state.offer))
+    tempOffer[event.target.name] = event.target.value
```
Comment (kvartborg) [inline src/views/Task/View.js]: This does not make sense?

File: src/views/Task/View.js [hunk 7]
```diff
@@ -1,64 +1,162 @@
...
+    this.state={
+      task:{
+        id:'',
+        start:'',
+        deadline:'',
+        urgency:'',
+        description:'',
+        title:''
+      },
+      offer:{
```
Comment (kvartborg) [inline src/views/Task/View.js]:
```suggestion
      estimate:{
```

File: src/views/Task/View.js (hunk 6 above)
Comment (kvartborg) [reply src/views/Task/View.js]:
```js
const tempEstimate = { ...this.state.estimate }
```

**Pull request merged**